pkg/apis/kops/cluster.go

```diff
@@ -452,12 +444,20 @@ type RBACAuthorizationSpec struct{}
 
 type AlwaysAllowAuthorizationSpec struct{}
 
-// AccessSpec provides configuration details related to kubeapi dns and ELB access
-type AccessSpec struct {
-	// DNS will be used to provide config on kube-apiserver ELB DNS
+// APISpec provides configuration details related to the Kubernetes API.
+type APISpec struct {
+	// DNS will be used to provide configuration for the Kubernetes API's DNS server.
 	DNS *DNSAccessSpec `json:"dns,omitempty"`
-	// LoadBalancer is the configuration for the kube-apiserver ELB
+	// LoadBalancer is the configuration for the Kubernetes API load balancer.
 	LoadBalancer *LoadBalancerAccessSpec `json:"loadBalancer,omitempty"`
+	// PublicName is the external DNS name for the Kubernetes API.
+	PublicName string `json:"publicName,omitempty"`
+	// InternalName is the internal DNS name for the Kubernetes API.
+	InternalName string `json:"internalName,omitempty"`
```

Comment:
I was always curious why we have `InternalName` as a field. It implies it can be changed.
Can `InternalName` be replaced by a function that always returns `"api.internal." + cluster.Name`?